doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,377 @@
+
+Capture real-time debugging data during model training in Amazon SageMaker
+==========================================================================
+
+To enable data emission for debugging model training, Amazon SageMaker initializes a "hook" which attaches itself to the training process and emits data necessary for debugging i.e. tensors. To provide the hook's configuration, specify the new option called ``DebuggerHookConfig`` when training your model. For more about the ``DebuggerHookConfig`` object, see the `API documentation <https://sagemaker.readthedocs.io/en/stable/debugger.html#sagemaker.debugger.DebuggerHookConfig>`__.
+
+The ``DebuggerHookConfig`` accepts one or more objects of type ``CollectionConfig`` which defines the configuration around a tensor collection you intend to emit and save during model training. A "collection" helps group tensors for easier handling.
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config = CollectionConfig(
+        name='collection_name',
+        parameters={
+            'key': 'value'
+        }
+    )
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission',
+        hook_parameters={
+            'key': 'value'
+        },
+        collection_configs=[
+            collection_config
+        ]
+    )
+
+    estimator = TensorFlow(
+        role=role,
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        debugger_hook_config=debugger_hook_config
+    )
+
+``hook_parameters`` in ``DebuggerHookConfig`` and ``parameters`` in ``CollectionConfiguration`` provide a way of fine tuning the configuration of the tensors stored during the training - they control the tensors that are saved and the step frequency at which they're saved.
+
+The possible values of ``hook_parameters`` in ``DebuggerHookConfig`` or ``parameters`` in ``CollectionConfig`` can be viewed at `CollectionParameters <https://docs.aws.amazon.com/sagemaker/latest/dg/API_CollectionConfiguration.html#SageMaker-Type-CollectionConfiguration-CollectionParameters>`__
+
+Specifying individual configuration for each collection
+-------------------------------------------------------
+
+``collection_parameters`` in the ``CollectionConfig`` object define what group of tensors are saved and how frequently they will be saved. This setting is specific to this collection.
+
+Specifying common configuration across all collections
+------------------------------------------------------
+
+If there are properties you want to apply across all collections, those can be supplied in ``hook_parameters`` within the ``DebuggerHookConfig`` object. For example, to apply a value of ``10`` for ``save_interval`` across all collections, do:
```

Comment:
you can remove "do" and leave it as just "to apply a value of `10` for `save_internal` across all collections:"

Reply:
Done.